### PR TITLE
Fix chan_shutdown_read issue

### LINKED ISSUE #7153: chan_shutdown_read error on remote host after connecting with OpenSSH
### Description

**What happened**:

Starting with 6.1, OpenSSH causes following error message on remote server after connecting: `channel 1: chan_shutdown_read: shutdown() failed for fd 8 [i0 o1]: Not a socket` Seems to trigger when you run `ssh-add` (or any variant of) on remote server after connecting.

**What you expected to happen**:

OpenSSH should work without triggering errors on remote client.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Connect to remote node via OpenSSH from a Windows client running 6.1+
2. Run `ssh-add` and observe error.

### Server Details
- Teleport version (run `teleport version`): 6.1+

### Client Details
- Tsh version (`tsh version`): 6.1+
- Computer OS (e.g. Linux, macOS, Windows): Windows

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!
`channel 1: chan_shutdown_read: shutdown() failed for fd 8 [i0 o1]: Not a socket`

### PR DESCRIPTION
**Pull request by EdwardDowling**
For graceful teardown, close the write part of the channel first. This
will send "EOF" packet (type 96) to the other side which will drain and
close the channel.

The regular close after that will send "close" packet (type 97) which
won't attempt to send us any more data since the channel is already
closed.

This mimics vanilla OpenSSH behavior. Without close_write first, the
agent client may be getting warnings like the following in stdout:

channel 1: chan_shutdown_read: shutdown() failed for fd 8 [i0 o1]: Not a socket

closes [7153](https://github.com/gravitational/teleport/issues/7153)

### CODE REVIEW

**Review by hugoShaka — APPROVED** [review]

**Comment by EdwardDowling** [comment]
@nklaassen could you have a look at this when you get a second?

File: lib/sshutils/ctx.go [hunk 1]
```diff
@@ -85,8 +85,23 @@ type agentChannel struct {
...
+// Close closes the agent channel.
 func (a *agentChannel) Close() error {
-	return a.ch.Close()
+	// For graceful teardown, close the write part of the channel first. This
+	// will send "EOF" packet (type 96) to the other side which will drain and
+	// close the channel.
+	//
+	// The regular close after that will send "close" packet (type 97) which
+	// won't attempt to send us any more data since the channel is already
+	// closed.
```
Comment (nklaassen) [inline lib/sshutils/ctx.go:96]: Could we add a test for this?

Comment (EdwardDowling) [reply lib/sshutils/ctx.go:96]: I've been trying to think of a nice way to test this properly but have not been able to come up with a good way to do this. Do you have any ideas how I could go about doing this?

Comment (EdwardDowling) [reply lib/sshutils/ctx.go:96]: I've added a test that will at least break if someone swaps these lines around,  still trying to figure out a nice way to mock everything required if we want to test the packets sent in the crypto/ssh package are what we expect

**Review by r0mant — APPROVED** [review]

File: lib/sshutils/ctx.go (hunk 1 above)
Comment (nklaassen) [reply lib/sshutils/ctx.go:96]: My original idea was that you could open an actual SSH connection and confirm the last 2 bytes sent, but your test at least confirms that Teleport closes things in the right order, so I think it's good enough as long as you've also confirmed the fix manually

**Review by nklaassen — APPROVED** [review]

**Pull request merged**